Task 1/2019201419王成瑞/double.h

```diff
@@ -0,0 +1,291 @@
+    return getExponent(&_val) == 0;
+}
+
+#define Rounding(PRECISION)                                             \
+        {                                                               \
+            int delta = (PRECISION);                                    \
+            uint128_t lowBits = frac_res & ((VL_1 << delta) - 1);       \
+            if(lowBits > VL_1 << (delta - 1)) {                         \
+                frac_res = (frac_res >> delta) + 1;                     \
+            } else if(lowBits < VL_1 << (delta - 1)) {                  \
+                frac_res >>= delta;                                     \
+            } else {                                                    \
+                frac_res >>= delta;                                     \
+                frac_res += (frac_res & 1);                             \
+            }                                                           \
+        }
+
+
+#define AdjustPrecision(PRECISION)                                      \
+        {                                                               \
+            while((frac_res >> (PRECISION)) >= 2) {                     \
+                exp_res += 1;                                           \
+                frac_res >>= 1;                                         \
+            }                                                           \
+            while((frac_res >> (PRECISION)) == 0) {                     \
+                frac_res <<= 1;                                         \
+                exp_res -= 1;                                           \
+            }                                                           \
+        }
+
+
+#define ProcessAnswer(PRECISION)                                        \
+        {                                                               \
+            if(exp_res > DOUBLE_EXP_MAX) {                              \
+                setInf(&result);                                        \
+            } else if(exp_res >= 1 - BIAS) {                            \
+                Rounding(PRECISION);                                    \
+                setExponent(&result, (unsigned)(exp_res + BIAS));       \
+                setFraction(&result, (uint64_t)frac_res);               \
+            } else if(exp_res >= 0 - BIAS - FRAC_SIZE) {                \
+                Rounding(PRECISION + 1 - BIAS - exp_res);               \
+                exp_res = -BIAS;                                        \
+                setExponent(&result, 0u);                               \
+                setFraction(&result, (uint64_t)frac_res);               \
+            } else {                                                    \
+                setZero(&result);                                       \
+            }                                                           \
+        }
+
+
+double_t __add__(double_t a, double_t b) {
```

Comment:
Don't declare your function name starting with double underscore since these names are reserved for the compiler, standard library or language itself. Refer to [What does double underscore ( __const) mean in C?](https://stackoverflow.com/questions/1449181/what-does-double-underscore-const-mean-in-c).

Reply:
发现并不能重载运算符之后就直接照搬了 python 的命名（逃